### ISSUE TITLE
Make indent spaces configurable

### ISSUE THREAD

**Issue by zdm** [ingydotnet/yaml-libyaml-pm#88]
Hi.
Currently it uses 2 spaces to indenting lines when generating yaml.
But it is hard to read, when yaml is large and has complex nested structure.
Is it possible to make this constant configurable from perl code?
My yaml looks much better with 4-spaces indentation.

**Comment by zdm** [comment]
You can insert this code to the "LibYAML/perl_libyaml.c", line 664, just after "yaml_emitter_initialize(&dumper.emitter);"

```
    /* set indent */
    SV* indent = get_sv("YAML::XS::IndentSpaces", 0);
    if (SvIOK(indent)) yaml_emitter_set_indent(&dumper.emitter, SvIV(indent));
```

It will take value from $YAML::XS::IndentSpaces or will use default (2 spaces).

**Comment by perlpunk** [comment]
Thanks!
The plan was to avoid more global variables and at some point use a better API, but that's still not happening in the near future.
@ingydotnet thoughts?

**Comment by zdm** [comment]
Oh, yes, you want to make it compatible with pure-perl YAMPL.pm, just rename var in code sample above to the YAML::Indent,

**Comment by zdm** [comment]
I fixed the code above and performed testing.
Now you can just insert it to the source and it will works.

**Comment by perlpunk** [comment]
@ingydotnet has to decide...

**Comment by ingydotnet** [comment]
I am ok with this but I think that in general we look at config vars in YAML::XS by looking at $YAML::XS::Indent and then $YAML::Indent. That way people can switch between YAML implementations and not need to change their settings.

Please confirm or deny that this is the way that config variables work for YAML::XS.

**Comment by zdm** [comment]
I will create pull request, where indent value will $YAML::XS::Indent // $YAML::Indent // 2.

**Comment by perlpunk** [comment]
@ingydotnet YAML::XS is currently only looking at variables in YAML::XS namespace

**Comment by perlpunk** [comment]
fixed by #89
